lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -502,9 +502,13 @@ def exec_module(self, **kwargs):
 
         if changed:
             if self.state == 'present':
-                subnet = self.get_subnet(virtual_network_resource_group, virtual_network_name, self.subnet_name)
-                if not subnet:
-                    self.fail('subnet {0} is not exist'.format(self.subnet_name))
+                subnet = self.network_models.SubResource(
```

Comment:
Surely there is a better way to do this than putting in the URL manually?

Reply:
this is a pattern i have "copied" from  azure_rm_loadbalancer. I think this is the only method which doesn't require actual query to Azure, which our customer wants to avoid, as the person using azure_rm_newtorkinterface doesn't have right to do this.